registry/remote/auth/client.go

```diff
@@ -302,7 +302,7 @@ func (c *Client) fetchBearerToken(ctx context.Context, registry, realm, service
 	if cred.AccessToken != "" {
 		return cred.AccessToken, nil
 	}
-	if cred == EmptyCredential || (cred.RefreshToken == "" && !c.ForceAttemptOAuth2) {
+	if cred == EmptyCredential {
```

Comment:
`ForceAttemptOAuth2` will be unused I think after this change. Which seems unusual, as presumably it exists for a reason?

Perhaps:
```suggestion
	if cred == EmptyCredential && !c.ForceAttemptOAuth2 {
```

(And to be clear, I'm speculating that if the user has set `c.ForceAttemptOAuth2 = true`, the user would expect the `c.fetchOAuth2Token(...)` to be invoked. Based on the naming only.)

Reply:
certainly, if `ForceAttemptOAuth2` will serve no function now (my proposal doesn't make sense), then should mark it deprecated

Reply:
I think from a security perspective alone, we should favor bearer auth. The current implementation favors basic auth.

Setting `ForceAttemptOAuth2` also makes it appear we aren't going to support basic auth if needed.

Why would we not allow by default grant type `password` using username/password?

Reply:
@TerryHowe This PR is becoming stale. Did you find anything suspicious?

Reply:
v1 has a backport of the same logic https://github.com/oras-project/oras-go/pull/89 but helm used the original https://github.com/oras-project/oras-go/tree/v1/pkg/auth it uses containerd which is what I was seeing https://github.com/oras-project/oras-go/blob/v1/pkg/auth/client.go#L23